### PR TITLE
Optimize interface for TA4 integration

### PR DESCRIPTION
**Pull request by anirban-chaudhuri**

### CODE REVIEW

**Comment by anirban-chaudhuri** [comment]
@SamWitty The demo notebook is breaking after the updates using sympy. I just wanted to check if it runs for everyone else and it might be a issue on my side. Can you do a quick check on this?

File: src/pyciemss/PetriNetODE/interfaces.py [hunk 1]
```diff
@@ -231,6 +231,321 @@ def load_and_calibrate_and_sample_petri_model(
...
+    *,
+    start_state: Optional[dict[str, float]] = None,
+    pseudocount: float = 1.0,
+    start_time: float = -1e-10,
+    method="dopri5",
+    verbose: bool = False,
+    n_samples_ouu: int = int(1e2),
+    maxiter: int = 2,
+    maxfeval: int = 25,
+) -> pd.DataFrame:
```
Comment (SamWitty) [inline src/pyciemss/PetriNetODE/interfaces.py:253]: This return type is not quite right anymore. Could you change this to `Tuple[pd.DataFrame, TODO]` where TODO is the type of the ouu_policy?

File: src/pyciemss/PetriNetODE/interfaces.py [hunk 2]
```diff
@@ -231,6 +231,321 @@ def load_and_calibrate_and_sample_petri_model(
...
+        n_samples_ouu: int
+            - The number of samples to draw from the model for each optimization iteration.
+        maxiter: int >= 0
+            - The maximum number of restarts for multi-start local optimization.
+            - maxiter = 0: leads to a single-start local optimization
+        maxfeval: int > 0
+            - The maximum number of function evaluations for each start of the local optimizer.
+
+    Returns:
+        samples: PetriSolution
```
Comment (SamWitty) [inline src/pyciemss/PetriNetODE/interfaces.py:303]: Could you add a description of the ouu_policy here?

**Review by SamWitty — CHANGES_REQUESTED** [review]
Looks great overall. I have a few minor comments on documenting the output of the optimize interface.

Could you also add a tiny smoke test for this interface? This should be very very easy, just add `self.optimized_samples` to the Test object similar to how it's done here (https://github.com/ciemss/pyciemss/blob/main/test/test_petrinet_ode/test_ode_interfaces.py#L52). Then add `self.optimized_samples` to each individual test where `self.samples` are used. E.g. https://github.com/ciemss/pyciemss/blob/main/test/test_petrinet_ode/test_ode_interfaces.py#L70

**Comment by anirban-chaudhuri** [comment]
> Looks great overall. I have a few minor comments on documenting the output of the optimize interface.
> 
> Could you also add a tiny smoke test for this interface? This should be very very easy, just add `self.optimized_samples` to the Test object similar to how it's done here (https://github.com/ciemss/pyciemss/blob/main/test/test_petrinet_ode/test_ode_interfaces.py#L52). Then add `self.optimized_samples` to each individual test where `self.samples` are used. E.g. https://github.com/ciemss/pyciemss/blob/main/test/test_petrinet_ode/test_ode_interfaces.py#L70

I added the test. I noticed that `self.intervened_samples` is also missing from the tests. Should I add that in?

File: src/pyciemss/PetriNetODE/interfaces.py (hunk 1 above)
Comment (anirban-chaudhuri) [reply src/pyciemss/PetriNetODE/interfaces.py]: Done

File: src/pyciemss/PetriNetODE/interfaces.py (hunk 2 above)
Comment (anirban-chaudhuri) [reply src/pyciemss/PetriNetODE/interfaces.py]: I have added description of the dictionary in the outputs.

**Review by SamWitty — APPROVED** [review]
Looks great!

**Pull request merged**